### PR TITLE
added a basic websocket server with some basic websocket communication

### LINKED ISSUE #2: Create a basic async websocket server

### PR DESCRIPTION
**Pull request by MaderNoob**

### CODE REVIEW

File: shithead_game_server/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    let mut game_server = GameServer::new()
+        .await
+        .context("failed to create game server")?;
+    game_server.start().await?;
+    Ok(())
+}
+
+#[tokio::main]
+async fn main() {
+    SimpleLogger::new()
```
Comment (nickBes) [inline shithead_game_server/src/main.rs:93]: Please explain what parameters and settings you're giving to the logger

File: shithead_game_server/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+use anyhow::Context;
+use futures::{SinkExt, StreamExt};
+use log::{error, warn};
+use simple_logger::SimpleLogger;
+use tokio::net::{TcpListener, TcpStream};
+use tokio_tungstenite::tungstenite::Message;
+
+const SERVER_BIND_ADDR: &str = "0.0.0.0:7522";
+
+pub struct GameServerState {}
```
Comment (nickBes) [inline shithead_game_server/src/main.rs:14]: Add a comment that explains the purpose of this class, as it wasn't implemented yet.

File: shithead_game_server/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+impl GameServerState {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+struct GameServer {
+    listener: TcpListener,
+    state: Arc<GameServerState>,
```
Comment (nickBes) [inline shithead_game_server/src/main.rs:28]: Add an explanation for this state.

File: shithead_game_server/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+) -> anyhow::Result<()> {
+    println!("client handler, addr: {}", addr);
+    let mut websocket = tokio_tungstenite::accept_async(stream)
+        .await
+        .context("failed to accept websocket client")?;
+    println!("done handling client");
+    while let Some(recv_result) = websocket.next().await {
+        let msg = recv_result.context("failed to receive message from client")?;
+        println!("received msg from client: {}", msg);
+        websocket
```
Comment (nickBes) [inline shithead_game_server/src/main.rs:74]: tell that this part is for testing and the game functionality will be implemented here.

File: shithead_game_server/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            .send(Message::text(format!("you said: {}", msg)))
+            .await
+            .context("failed to send message to client")?;
+    }
+    println!("done handling client");
```
Comment (nickBes) [inline shithead_game_server/src/main.rs:79]: Also add tell that this log is for debug.

**Review by nickBes — CHANGES_REQUESTED** [review]
Nice, though need more explenations

File: shithead_game_server/src/main.rs (hunk 5 above)
Comment (MaderNoob) [reply shithead_game_server/src/main.rs:79]: This is for testing purposes only, will be removed as soon as we move on

**Review by nickBes — APPROVED** [review]
Really good!

**Pull request merged**